packages/aws-rfdk/lib/lambdas/nodejs/version-provider/test/handler.test.ts

```diff
@@ -3,14 +3,21 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/* eslint-disable dot-notation */
+/* eslint-disable dot-notation, no-console */
 
 import { VersionProviderResource } from '../handler';
 
 describe('VersionProviderResource', () => {
   let versionProviderResource: VersionProviderResource;
+  const consoleLog = console.log;
   beforeEach(() => {
     versionProviderResource = new VersionProviderResource();
+    // tslint:disable-next-line: no-empty
+    console.log = () => {};
+  });
+
+  afterEach(() => {
+    console.log = consoleLog;
   });
```

Comment:
nit: jest includes a nice utility for monkey-patching existing modules/APIs. Not critical, but it would avoid the extra `consoleLog` variable and potentially the `no-console` eslint disabling since this would get delegated to jest.

```ts
beforeEach(() => {
  // ...
  jest.spyOn(console, 'log').mockReturnValue(undefined);
  // ...
});

afterEach(() => {
  // ...
  jest.clearAllMocks();
  // ...
});
```

What do you think?

Reply:
iirc, I tried this in the tests for the mongodb handler (ex: here -- https://github.com/aws/aws-rfdk/blob/db0e5b93df76bc1257124e9033e01c1de6964299/packages/aws-rfdk/lib/core/lambdas/nodejs/mongodb/test/handler.test.ts#L363 ), but there were some issues with it; the details of which I do not recall. The only way that I could get it to work was the console log trick that David is doing here.

Reply:
I had copied this from the file Daniel mentioned, but I'm always down for learning new things, so I tried out the spies like Josh suggested and they worked for me. I tried them out in the Mongo tests too, and I was able to get them to work there, so I updated them as well.